pull.go

```diff
@@ -69,8 +92,11 @@ func (c *Client) Pull(ctx context.Context, ref string, opts ...RemoteOpt) (Image
 	i := NewImageWithPlatform(c, img, pullCtx.PlatformMatcher)
 
 	if pullCtx.Unpack {
-		if err := i.Unpack(ctx, pullCtx.Snapshotter, pullCtx.UnpackOpts...); err != nil {
-			return nil, errors.Wrapf(err, "failed to unpack image on snapshotter %s", pullCtx.Snapshotter)
+		// Unpack for schema 1 image.
+		if i.Target().MediaType == images.MediaTypeDockerSchema1Manifest {
```

Comment:
The target shouldn't have this media type. Instead is there a way to just see if the simultaneous unpacker didn't perform any unpacks, then try and do it synchronously

Reply:
Oops, the image was converted already. My bad.

Let me see.

Reply:
You can use an atomic counter on the number of unpack calls

Reply:
Done. Added an atomic int `unpacks`.